actions/check-thirdparty-notice/action.yml

```diff
@@ -0,0 +1,76 @@
+# SPDX-FileCopyrightText: Copyright (C) 2025 Opal Health Informatics Group <https://www.opalmedapps.com>
+#
+# SPDX-License-Identifier: MIT
+
+# Ensures that declared project dependencies and the third-party notice in THIRDPARTY.md are in sync.
+# I.e., each dependency has to have a corresponding heading in the notice.
+# Warns about dependencies missing as well as dependencies that are declared in addition.
+#
+# Note: Currently supports npm, composer and pip.
+
+name: Check Thirdparty Notice
+description: Ensures that declared project dependencies and the third-party notice in THIRDPARTY.md are in sync.
+
+inputs:
+  package-file:
+    description: The filename of an npm package.json file (can be relative to the current working directory)
+    default: ""
+  composer-file:
+    description: The filename of a composer.json file (can be relative to the current working directory)
+    default: ""
+  pip-requirements-files:
+    description: |
+      The filename(s) of a requirements file (can be relative to the current working directory).
+      If specifying more than one file, separate with spaces.
+    default: ""
+  additional-dependencies:
+    description: |
+      Additional dependencies that are allowed to be listed in the third-party notice.
+      If specifying more than one package, separate with spaces.
```

Comment:
I haven't figured this out myself yet: Somehow it must be possible to provide an array as input, e.g.,

```yaml
additional-dependencies:
  - foo
  - bar
```

Maybe something we can figure out later, I think it would be better than separating by space.

Reply:
Note: we determined that this would be nice, but it might not be possible.